torch/distributed/pipelining/stage.py

```diff
@@ -719,7 +716,8 @@ def forward_one_chunk(
         output_tuple = _normalize_model_output_as_tuple(output)
 
         # Prepare for final output merge or reduction
-        self.output_chunks.append(output)
+        if self.is_last:
+            self.output_chunks.append(output)
```

Comment:
nit: do you mind adding more comments here re why only the last stage needs it? Basically what's stated in the PR desc. Thanks!